### DISCUSSION TITLE
Pro Microを基板に載せるとパソコンから認識されません / 原因を探りたいです

### DISCUSSION THREAD

**Discussion by megmilk** [Yowkees/keyball#381] [Q&A] [answered]
お世話になっております

Keyball39を組み立て、テスト用のファームウェアを書き込みました

# ご相談

- どこに原因があるか探りたいです
- https://github.com/Yowkees/keyball/discussions/349 に写真があったようですが私からは見えず

# 症状

- 右側がパソコンから認識されません
- https://github.com/Yowkees/keyball/discussions/349 の症状とよく似ている

## 左側
- 難なくファームウェアの書き込みができました
- すべてのLED点灯しました

## 右側
- Pro Microを基板に載せた状態では
  - パソコンから認識されません
  - そのためファームウェアの書き込みができません
  - いちおうPro MicroのLEDは赤く点灯しています
  https://github.com/Yowkees/keyball/discussions/349 の症状と似ているがPro MicroのLEDは消灯しない
- Pro MicroをKeyball39の基板から離し、Pro Micro単体にすると
  - パソコンから認識されます
  - ファームウェアの書き込みもできます
- ファームウェアを書き込んだPro MicroをKeyball39の基板に載せると、やはりパソコンからは認識されません
- その他の症状として、自分ではんだ付けしたLEDがまったく点灯しません

## その他試したこと
- 左側のPro Microを右側の基板に載せると、やはりパソコンから認識されません
- 右側のPro Microを左側の基板の載せると、パソコンから認識されます

# 以上のことから
右側の基板のどこかに異常があるように思います

ご面倒をおかけいたします
よろしくお願いいたします

**Comment by megmilk** [comment]
お騒がせしました

どうやら右側基板のはんだ付け不良だったようです
もう一度つけ直す感覚で、すべてのはんだ付け箇所にはんだごてを当てたところ改善しました

ただ、何度も試行錯誤するうちにPro Microが1つ壊れてしまったようです

- Pro Micro単体をパソコンに繋いでも認識せず
- Pro Micro本体の赤色LEDが1つだけ点灯しています
- Pro Micro単体でパソコンに接続していますが、LEDの光が瞬くような光りかたをしています

どうにも直らなれば、新しいPro Microを買います

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

ご自身で半田不良を解決していただき、誠にありがとうございました。
ProMicroの不良については単体でもPCで認識されないということで、やはり故障の可能性が濃厚でしょうか。
半田付けで隣とのショート等ご確認いただき、解決しないようでしたら交換を検討していただければ幸いです。
なお、Keyballシリーズご購入者様限定で、12Pピンヘッダー単体での販売も可能ですので、ご検討ください。

以上、お手数ですがご対応お願いいたします。

**Comment by megmilk** [comment]
@Yowkees 返信ありがとうございます！

そうですね、Pro Micro の復旧方法についてインターネットから情報を探すのですが、私のような症状は見つかりませんでした

おそらく何度もソケットに抜き差ししたり、Pro Micro単体でリセットとGNDを短絡させるときに他にピンが触れてしまい壊してしまったのだと思います

なので遊舎工房さんで同じPro Microを頼んでおきました

また休日に組んでお知らせします

> Keyballシリーズご購入者様限定で、12Pピンヘッダー単体での販売も可能です

そうだったんですね、これは知りませんでした！

手間になってしまうので、Pro Microを買ったときに少し長めのピンヘッダが付属していたので、適度にカットして使おうかと思っています

**Comment by megmilk** [comment]
@Yowkees  お世話になっております

右側のPro Microを交換してようやくKeyball39が完成しました！

いろいろとお騒がせしました

完成したのですが、気になっていることがございます

# 現状
Pro Microに付属のピンヘッダを使用したところ

![PXL_20231014_033010590 のコピー](https://github.com/Yowkees/keyball/assets/54742534/642060ac-b584-4a9c-a501-86aadc8199f7)

(1) 少し厚みがあります
(2) これによりLCDや保護用のアクリル板が斜めになります

厚みに違いがあるとは知らず誤算でした

すぐに支障があるわけではありませんが、ソケットとLCDの接続が不安定です
私としては真っ直ぐな状態で使用したいです

# 直す方法 (案)

## (その１) Pro Microと樹脂部分の薄いピンヘッダを購入

- 一度はんだ付けしたピンヘッダを外しての交換は困難だと考えます

## (その２) LCDのピンヘッダを少し長めのものに交換

- さらに長めのスペーサーに交換または、ワッシャー等を挟み、長さを延長
- さらに別途少し長めのネジを探す

前者の方法が最善だと私は考えています
後者は部品を探す手間が非常にかかると思います

もちろん後者の方法についてなにかご存知でしたら返信いただければと思います

# ピンヘッダ単品での注文

以前にご教示いただいたこの話ですが

> Keyballシリーズご購入者様限定で、12Pピンヘッダー単体での販売も可能です

こちらはどのような方法で購入できますか？

ちなみに私はKeyball39は遊舎工房さんで購入しました

よろしくお願いいたします

**Comment by Yowkees** [comment, marked as answer]
ご対応ありがとうございます。

OLED保護プレートが斜めになっている件について、
一番簡単なのが、ご提案の通り、長めのスペーサーに交換することだと思います。
キットに入っているのは11mmのもので、型番が「ARB-2011E」という商品で、ヒロスギネット等で購入可能です。
これの1mm～3mm長いものをご検討ください。

次にキットに含まれているピンソケット・ピンヘッダー等をご購入希望の場合は、下記の白銀ラボウェブショップのお問合せよりご連絡ください。
https://shirogane-lab.net/inquiry

以上、ご検討よろしくお願いいたします。

**Reply by megmilk** [reply]
ご返信ありがとうございます！

なるほど、スペーサーもいろいろあるのですね！

ご案内いただいた問い合わせフォームに、当質問の続きを送らせていただきました

一旦、この質問は終了とさせていただきます

ありがとうございます！
